PIL/ImageFile.py

```diff
@@ -95,22 +98,11 @@ def __init__(self, fp=None, filename=None):
 
         try:
             self._open()
-        except IndexError as v:  # end of data
-            if Image.DEBUG > 1:
-                traceback.print_exc()
-            raise SyntaxError(v)
-        except TypeError as v:  # end of data (ord)
-            if Image.DEBUG > 1:
-                traceback.print_exc()
-            raise SyntaxError(v)
-        except KeyError as v:  # unsupported mode
-            if Image.DEBUG > 1:
-                traceback.print_exc()
-            raise SyntaxError(v)
-        except EOFError as v:  # got header but not the first frame
-            if Image.DEBUG > 1:
-                traceback.print_exc()
-            raise SyntaxError(v)
+        except (IndexError,  # end of data
+                TypeError,  # end of data (ord)
+                KeyError,  # unsupported mode
+                EOFError) as v:  # got header but not the first frame
+            logger.exception("%s")
```

Comment:
This is a change in functionality. It used to raise SyntaxError. Does `logger.exception()` raise anything? If not, it's a change in functionality. If it does, it's a different error reason.